stylix/home-manager-integration.nix

```diff
@@ -0,0 +1,80 @@
+homeManagerModule:
+{ lib, config, options, ... }:
+
+let
+  alwaysCopy = path: {
+    inherit path;
+    condition = lib.const true;
+  };
+
+  copyModules = builtins.map
+    (
+      { path, condition }:
+      { config, osConfig, ... }:
```

Comment:
I always thought `builtins.map` receives only one argument. Could you enlighten me where this second argument comes from, as I don't see where those `config` and `osConfig` arguments come from?

Reply:
The `builtins.map` call is generating a list of Home Manager modules of the following format:

```nix
[
  ({ config, osConfig, ... }: lib.mkIf (condition config) {
    stylix.image = lib.mkDefault osConfig.stylix.image;
  })
  ({ config, osConfig, ... }: lib.mkIf (condition config) {
    stylix.base16Scheme = lib.mkDefault osConfig.stylix.base16Scheme;
  })
  ...
]
```

So the `{ config, osConfig, ... }` argument is what would normally be the first line of a `hm.nix`.

`config` is the Home Manager configuration and `osConfig` is either the NixOS configuration or the nix-darwin configuration, depending on the OS.